Neos.Neos/Classes/FrontendRouting/EventSourcedFrontendNodeRoutePartHandler.php

```
@@ -404,6 +428,32 @@ private function buildResolveResultFromUri(UriInterface $uri): ResolveResult
         return new ResolveResult($uri->getPath(), $uriConstraints);
     }
 
+    /**
+     * Whether the given node is allowed according to the "nodeType" option and the "onlyMatchSiteNodes" option.
+     */
+    private function nodeTypeIsAllowed(
+        DocumentNodeInfo $nodeInfo,
+        ContentRepository $contentRepository,
+    ): bool
+    {
+        $nodeTypeManager = $contentRepository->getNodeTypeManager();
+
+        $allowedNodeTypeName = $this->options['nodeType'] ?? null;
+        if ($allowedNodeTypeName && !$nodeTypeManager
+            ->getNodeType($nodeInfo->getNodeTypeName())
+            ?->isOfType($allowedNodeTypeName)) {
+            return false;
+        }
+
+        $onlyMatchSiteNodes = $this->options['onlyMatchSiteNodes'] ?? false;
+        if ($onlyMatchSiteNodes && !$nodeTypeManager
+                ->getNodeType($nodeInfo->getNodeTypeName())
+                ?->isOfType('Neos.Neos:Site')) {
+            return false;
+        }
+        return true;
```

I think the implementation is quite clever because it allows both `nodeType` and `onlyMatchSiteNodes` to be configured and it will only match nodes that satisfy _both_.
But I wonder: Is that actually the behavior that we want to enforce or should we throw an exception if both are configured for one route?

---

I first had a simpler way and then realised I should have the identical behaviour to the 8.3 variant which also supported both options at the same time.